utils/event-publisher/src/event_layer.rs

```diff
@@ -0,0 +1,58 @@
+use mbus_api::event::EventMessage;
+use std::fmt::Debug;
+use tokio::sync::mpsc::{error::TrySendError, Sender};
+use tracing::field::{Field, Visit};
+use tracing_subscriber::{layer::Context, Layer};
+
+pub struct EventLayer {
+    sender: Sender<EventMessage>,
+}
+
+impl EventLayer {
+    pub fn new(sender: Sender<EventMessage>) -> Self {
+        Self { sender }
+    }
+}
+
+impl<S> Layer<S> for EventLayer
+where
+    S: tracing::Subscriber,
+{
+    fn on_event(&self, event: &tracing::Event<'_>, _ctx: Context<'_, S>) {
+        let mut visitor = EventVisitor::default();
+
+        event.record(&mut visitor);
+
+        let output = visitor.0;
+
+        if let Err(err) = self.sender.try_send(output) {
+            match err {
+                TrySendError::Full(_) => {
+                    tracing::warn!("Buffer is full");
```

Comment:
when can this happen, can we limit these warnings to once every X seconds?

Reply:
Whenever there is a new message to be sent to the buffer and the buffer is full this error occurs. I am just printing out the error without interrupting the code. Should I limit the warnings based on timings?

Reply:
Yes I think so, to avoid flooding stdout with errors if nats/buffer get full.